### ISSUE TITLE
People fall detection

### ISSUE THREAD

**Issue by ivelin** [ambianic/ambianic-edge#182]
This issue is work in progress. Looking for help to wrap it up. 
[Draft PR here](https://github.com/ambianic/ambianic-edge/pull/200).

Background:
- COVID19 isolated many home care patients without access to their caretakers. https://devpost.com/software/remote-elderly-home-care-via-privacy-preserving-surveillance-y8sgxi

- Hospice care and home care are long term growth markets with insufficient nursing resources available to attend to elderly care receivers.

- Falls are a major indicator of more serious health issues and cause of complications.

- Prior research:
https://arxiv.org/abs/1711.11200

- NNStreamer suggests one possible solution (at minute 23):
https://gstconf.ubicast.tv/permalink/v125ac31257ffai03d34/#start=1392

- Another promising direction to explore is with the recently published TFLite pose detection model plus simple heuristics measuring the relation of head coordinates to hips and knees.

Related resources:
https://github.com/reigngt09/Pose-Estimation/tree/master/3.%20Fall%20Detection
https://www.youtube.com/watch?v=IlsXQPOF9IE
https://www.tensorflow.org/lite/models/pose_estimation/overview

Further in this issue there are comments with links to publicly available training data and other helpful resources.

**Comment by njcurtis3** [comment]
I'd like to also be on this one if possible. I will stick with these two assignments on the Agile board for now. That way I can stay focused. I will still be participating in all of the chat threads in all of the other areas.

**Comment by ivelin** [comment]
Sounds good. I am sending you team member invitation , so you can claim and control issues in the future.

**Comment by ivelin** [comment]
Another potential resource to consider for classifying falls in photos:

https://blog.tensorflow.org/2020/05/bigtransfer-bit-state-of-art-transfer-learning-computer-vision.html

**Comment by ivelin** [comment]
Pull request draft: https://github.com/ambianic/ambianic-edge/pull/200

**Comment by ivelin** [comment]
Google AI open sourced Pose Detection 2.0 based on Resnet 50
https://github.com/tensorflow/tfjs-models/tree/master/posenet

**Comment by ivelin** [comment]
Google AI open sourced the pose detection resnet models for TFLITE and Coral with Python code:
https://github.com/google-coral/project-posenet

**Comment by ivelin** [comment]
Great browser demo to experiment with various body positions - standing, sitting, laying down.
https://storage.googleapis.com/tfjs-models/demos/posenet/camera.html

**Comment by ivelin** [comment]
Public study with video samples of elderly falls: https://www.thelancet.com/journals/lancet/article/PIIS0140-6736%2812%2961263-X/fulltext#relVideo
5 examples of elderly falls: https://www.youtube.com/watch?v=p5i4z3sNaKM

Other public sources:
https://www.youtube.com/watch?v=JDxBAiUuQus
https://www.youtube.com/watch?v=mCpAdH_zkIA
https://www.youtube.com/watch?v=f39ednyQMhg
https://www.youtube.com/watch?v=mCpAdH_zkIA

**Comment by ivelin** [comment]
Public fall detection datasets for ML:

http://sites.google.com/up.edu.mx/har-up/
https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6539235/

https://github.com/AdrianNunez/Fall-Detection-with-CNNs-and-Optical-Flow#extracted-features-and-labels

https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6832422/#app1-sensors-19-04565

https://archive.ics.uci.edu/ml/datasets/Simulated+Falls+and+Daily+Living+Activities+Data+Set

https://www.kaggle.com/pitasr/falldata?select=falldeteciton.csv

http://le2i.cnrs.fr/Fall-detection-Dataset?lang=en

http://videodatasets.org/

http://fenix.univ.rzeszow.pl/~mkepski/ds/uf.html

http://www.iro.umontreal.ca/~labimage/Dataset/

http://falldataset.com/

**Comment by ivelin** [comment]
Quick status update on this issue. We are almost there:
https://github.com/ambianic/ambianic-edge/pull/200#issuecomment-683998087

**Comment by ivelin** [comment]
Progress update: https://github.com/ambianic/ambianic-edge/pull/200#issuecomment-692231125

**Comment by ivelin** [comment]
Update Nov 30, 2020. @bhavikapanara , @muka, John Senay and I had a catch up call. Summary:

1. We will not worry about the Coral version. Focus on tflite and CPU test cases.
2. Test newer resnet50 model, which seems to show better results than v1 mobilenet. Demo page to play with options.
3. We need at least 1fps performance on rpi4. 2fps is better to give us some CPU slack. More than 2fps is overkill for our use case. Let’s pick an optimal posenet ml model and image size (tensor input) that fits this performance criteria.
4. The initial empirical fall detection will compare the angle between the line drawn from shoulder to hip (whichever side is visible: left-left or right-right). If the angle of this line between two frames that are 1-2 seconds apart is more than 60% we assume its a fall alert.
5. For test cases, we can use any reasonable dataset, including home made. Rich public data sets available here for future model tuning.
6. Once we complete 1-5, we can begin discussion about a custom build DNN ML model that focuses on fall detection. Some options are classification between images or an RNN (e.g. transformer, bert) that predicts falls based on a continuous sequence of poses.

**Comment by ivelin** [comment]
Closed by https://github.com/ambianic/ambianic-edge/pull/266
